controller/appcontroller.go

```diff
@@ -1331,6 +1331,7 @@ func (ctrl *ApplicationController) setOperationState(app *appv1.Application, sta
 			logCtx.Warnf("error patching application with operation state: %v", err)
 			return fmt.Errorf("error patching application with operation state: %w", err)
 		}
+		ctrl.writeBackToInformer(patchedApp)
```

Comment:
Instead of sprinkling in `ctrl.writeBackToInformer()` everwhere that we call PatchApp, maybe we can create a helper method that does both the Patch + writeback in order to keep this DRY.

I do understand we could only do this for Patch() since we have one Update() call that could not use it.

Reply:
Updated code with a wrapper for patch + writeback.